### DISCUSSION TITLE
CSI capture and Extraction

### DISCUSSION THREAD

**Discussion by Mweme** [seemoo-lab/nexmon_csi#240] [Q&A] [answered]
Hello team,
I'm new to the channel thing, but currently is the place I'm working my research.
I have installed nexmon in RPi and was successful and I have configured another RPi to work as AP.
But I would like to have guidance on the following:-

How do I capture CSI frames?
How do I visualize these frames?
Which file do these CSI frames are kept in Rx?
How analysis has been done?
Any assistance will be much appreciated.

**Comment by zeroby0** [comment, marked as answer]
I know it isn't much detailed, I'm a bit busy these day, but here are some quick pointers.

> How do I capture CSI frames?

Set your transmitter Pi to transmit in channel 36 at 40 MHz, then use makecsiparams and generate a configuration string.
makecsiparams -c 36/40 -C 1 -N 1

makecsiparams has to be compiled with `make`, it's in `utils/makecsiparams`. 

`makecsiparams -c 36/40 -C 1 -N 1` captures CSI of ALL packets going on[ channel 36](https://en.wikipedia.org/wiki/List_of_WLAN_channels), to capture only packets from your Pi, use the -m flag with transmitter Pi's mac address (usage in readme)

This tries to capture all packets coming from your transmitter Pi, if you need only some packets, use the `-b` flag and filter by [frames first byte](https://en.wikipedia.org/wiki/802.11_Frame_Types)

Then follow the readme and setup CSI collection. Use tcpdump to collect packets, the packets contain CSI in them, and store them as pcap.

I have a script that does most of the work for you, and some documentation here: https://github.com/nexmonster/nexmon_csi/tree/pi-5.4.51

> How do I visualize these frames?

There is a matlab script in the utils folder, but if you want to use Python, use this: https://github.com/nexmonster/nexmon_csi/tree/feature/python/utils/python

I'm planning to release it as a separate tool, but till then, you can use this. You can also use [CSIkit](https://github.com/Gi-z/CSIKit) by @Gi-z.

> Which file do these CSI frames are kept in Rx?

Whichever file you write to with `tcpdump -o` 

> How analysis has been done?

Depends on how you want to use the CSI values. You will probably have to figure this out on your own.

Make sure you're using a supported kernel version. The latest version is not supported yet, so if you download the latest raspberry os, it won't work. Link to the latest raspberry os with a supported kernel is in my fork: https://github.com/nexmonster/nexmon_csi/tree/pi-5.4.51
